.github/workflows/ci.yml

```diff
@@ -260,3 +260,106 @@ jobs:
       # when imported in an NPM node_modules tree.
       - name: Prepack packages
         run: yarn lerna run --reject-cycles --concurrency 1 prepack
+
+  test-xs:
+    name: test-xs
+
+    # begin macro
+
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [20.x]
+        moddable-version: [5.0.0]
+        platform: [ubuntu-latest]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # without this, setup-node errors on mismatched yarn versions
+      - run: corepack enable
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: yarn
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      # end macro
+
+      - name: Run yarn build
+        run: yarn build
+
+      - name: Get XS
+        run: |
+          wget https://github.com/Moddable-OpenSource/moddable/releases/download/${{ matrix.moddable-version }}/xst-lin64.zip
+          mkdir -p bin
+          unzip xst-lin64.zip -d bin
+          chmod 755 bin/xst
+
+      - name: Is this thing on?
+        run: |
+          PATH=$PATH:$GITHUB_WORKSPACE/bin
+          xst -e true && ! xst -e null.reference
+
+      - name: Run XS tests
+        run: |
+          PATH=$PATH:$GITHUB_WORKSPACE/bin
+          yarn lerna run test:xs
+
+  # for testing a version with a particular hash, when pre-built binaries are unavailable
+  # make-and-test-xs:
+  #   name: make-and-test-xs
+  #
+  #   # begin macro
+  #
+  #   runs-on: ${{ matrix.platform }}
+  #   strategy:
+  #     fail-fast: false
+  #     matrix:
+  #       moddable-version: []
+  #       node-version: [20.x]
+  #       platform: [ubuntu-latest]
+  #
+  #   steps:
+  #     - name: Checkout
+  #       uses: actions/checkout@v3
+  #
+  #     # without this, setup-node errors on mismatched yarn versions
+  #     - run: corepack enable
+  #
+  #     - name: Use Node.js ${{ matrix.node-version }}
+  #       uses: actions/setup-node@v3
+  #       with:
+  #         node-version: ${{ matrix.node-version }}
+  #         cache: yarn
+  #
+  #     - name: Install dependencies
+  #       run: yarn install --immutable
+  #
+  #     # end macro
+  #
+  #     - name: Run yarn build
+  #       run: yarn build
+  #
+  #     - name: Clone XS
+  #       run: git clone https://github.com/moddable-OpenSource/moddable --branch "${{ matrix.moddable-version }}" --depth 1
```

Comment:
I'd recommend using `actions/checkout`, also avoid to checkout as a subfolder of the repo under testing, it tends to complicate things